### ISSUE TITLE
Cannot fragment Replicate structs larger than 508 bytes

### ISSUE THREAD

**Issue by DarkZek** [naia-lib/naia#67]
Hi, I love the asset and it has been working pretty well so far! It is very impressive.

I have found an issue with large packets, webrtc-unreliable lists a max size of ~1200 bytes which is what is used so I would assume my ~600 byte packet should work. This is reproducible in the bevy example by changing entity_assignment in the shared crate to contain my large amount of data.
```#[derive(Component, Replicate)]
#[protocol_path = "crate::protocol::Protocol"]
pub struct EntityAssignment {
    pub entity: EntityProperty,
    pub assign: Property<bool>,
    pub data: Property<[[[u8; 8]; 8]; 8]>
}

impl EntityAssignment {
    pub fn new(assign: bool) -> Self {
        EntityAssignment::new_complete(assign, [[[0; 8]; 8]; 8])
    }
}
```

The next time send_all_updates runs it will not return.

**Comment by DarkZek** [comment]
I see this is related to the [contants limiting size](https://github.com/naia-lib/naia/blob/main/shared/src/constants.rs). There definitely should be an error once this is hit (possibly only in debug mode)

**Comment by connorcarpenter** [comment]
Hi DarkZek, thanks for logging an issue :)

Yes, each packet `naia` sends will need to be less than 508 bytes, as you discovered. `naia` can usually fragment large messages down into a bunch of tiny packets, but it can't currently fragment a single `Replicate` struct.

This is a feature that must be added. I'm going to rename this issue if that's alright

**Comment by BrandonDyer64** [comment]
Is it possible to send large messages by not using `Replicate`?

**Comment by connorcarpenter** [comment]
This is still an issue with `naia` .. working to resolve it soon with automatic Packet Fragmentation. However with Release 0.15, no longer will the app just seize up and stop sending packets - `naia` will throw a helpful error message to help you when going over the 508 byte limit

**Comment by connorcarpenter** [comment]
Fixed / implemented with Release 0.18: https://github.com/naia-lib/naia/releases/tag/v0.18.0
